From cd23d28c98147706c41a7b9b6ea319f165541210 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 12 Dec 2024 16:05:51 +0800
Subject: [PATCH 248/260] AOSCOS: MIPS: select ARCH_FORCE_MAX_ORDER >= 11 if
 NUMA_BALANCING is selected

This should suppress an assertion failure during build. Further
investigation should be conducted when possible.

Fixes: "BACKPORT: FROMLIST: mips/mm: Add NUMA balancing support"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 1452041deae6..7d12850fea04 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -2112,9 +2112,9 @@ config ZBOOT_LOAD_ADDRESS
 
 config ARCH_FORCE_MAX_ORDER
 	int "Maximum zone order"
-	default "13" if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_64KB
-	default "12" if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_32KB
-	default "11" if MIPS_HUGE_TLB_SUPPORT && PAGE_SIZE_16KB
+	default "13" if (NUMA_BALANCING || MIPS_HUGE_TLB_SUPPORT) && PAGE_SIZE_64KB
+	default "12" if (NUMA_BALANCING || MIPS_HUGE_TLB_SUPPORT) && PAGE_SIZE_32KB
+	default "11" if (NUMA_BALANCING || MIPS_HUGE_TLB_SUPPORT) && PAGE_SIZE_16KB
 	default "10"
 	help
 	  The kernel memory allocator divides physically contiguous memory
-- 
2.48.1

